root/etc/e-smith/events/actions/nethserver-dc-user-modify

```diff
@@ -40,3 +45,31 @@ if ($? != 0) {
     die("[ERROR] User $userName modification failed\n");
 }
 
+
+#Create a ldif file and use it to modify the shell access
+#a random name and a chmod is done for security
+my $path = '/var/lib/machines/nsdc/var/lib/samba/private/';
+my $fileName = "$userName-". sprintf("%08X", rand(0xFFFFFFFF)).".ldif";
+my $filePath = $path . $fileName;
+
+my $contentFile = qq (
+dn: cn=$userName,cn=Users,$baseDN
+changetype: modify
+replace: loginShell
+loginShell: $shell
+);
+
+open(my $fh, '>', $filePath) or die "Could not open file '$filePath' $!";
+chmod 0600, $filePath;
```

Comment:
In general the most secure approach is setting `umask` appropriately _before_ the file creation, instead of _chmod_ it at a later step